src/sdl/UserEvent.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+   Copyright (C) 2017 the Battle for Wesnoth Project http://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <SDL_events.h>
+
+class Wesnoth_SDL_UserEvent : public SDL_UserEvent
+{
+    public:
+        Wesnoth_SDL_UserEvent() {
+            memset (this, 0, sizeof(SDL_UserEvent));
+        }
+
+        Wesnoth_SDL_UserEvent(int type) : Wesnoth_SDL_UserEvent() {
+            this->type = type;
+        }
+
+        Wesnoth_SDL_UserEvent(int type, int code) : Wesnoth_SDL_UserEvent(type) {
+            this->code = code;
+        }
+
+        Wesnoth_SDL_UserEvent(int type, int data1, int data2) : Wesnoth_SDL_UserEvent(type) {
+            this->data1 = reinterpret_cast<void*>(data1);
+            this->data2 = reinterpret_cast<void*>(data2);
+        }
+
+        Wesnoth_SDL_UserEvent(int type, void* data1) : Wesnoth_SDL_UserEvent(type) {
+            this->data1 = data1;
+        }
+};
```

Comment:
You forgot the trailing newline.

Reply:
Ah.  And there are artifacts from earlier versions, too.

I'll push an update for all the comments tomorrow.